crates/bench-utils/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+//! Helper utilities shared across the Criterion benches in this workspace.
+//!
+//! The worker benches all follow the same pattern:
+//! 1. Spawn the server binary under test with [`start_server`].
+//! 2. Probe the server until it is ready using [`warm_up_server`].
+//! 3. Drive Criterion against one or more HTTP endpoints via [`benchmark_server`].
+//! 4. Tear the server back down with [`stop_server`].
+//!
+//! Keeping these helpers here avoids copy‑pasting the bootstrapping logic
+//! into every benchmark file and makes it easy to introduce new benches—
+//! import this crate, call into the helpers, and focus on the benchmarked
+//! request/response logic itself.
+
+use criterion::Criterion;
+use reqwest::Client;
+use std::{
+    process::{Child, Command, Stdio},
+    time::Duration,
+};
+use tokio::{runtime::Runtime, time::sleep};
+
+/// Compose a benchmark URL from an address stored in an env var.
+///
+/// The benches set (for example) `WORKER_ADDR=127.0.0.1:4000` before running.
+/// Passing `("WORKER_ADDR", "health")` yields `http://127.0.0.1:4000/health`.
+pub fn get_url(env_key: &str, endpoint: &str) -> String {
```

Comment:
suggestion: I find a combination of envy and dotenv crates extremely handy for your env lookups.  It allows you to default variables if you want or use .env files.